### PR TITLE
Bug Fixes from Eatery Launch

### PR DESCRIPTION
**Pull request by vinnie4k**
## Overview

Fixed some bugs and UI improvements mentioned in GitHub issues.

### Issues Addressed

- Addressed **Incorrect times displayed on homepage #90**
- Addressed **favoriting on home screen not auto load #83**
- Addressed **View eatery details infinite load glitch #66**
- Addressed **Home page favorites fail to update properly**
- Addressed **Home page filters not working as expected #78**
- Addressed **********Upcoming menus expanding card glitch #65**********
- Addressed **104 West says it opens until 5PM on the homepage, but details page displays 2PM #72**
- Addressed **[Performance] Image caching on homepage #75**
- Addressed **The default is all campus, but if I just want west dining halls and toggle west, it only shows north and central (due to wrong filtering) #88**
- Addressed **[UX Request] Favorite eateries #73**
- Addressed **Screen navigation glitches out on logout #67**
- Addressed **Upon login, loading spinner never stops spinning #70**
- Addressed **Web browser can be closed during log in #69**
- Addressed **nearest to you section doesn't display on launch. it takes ~10 sec to load and display. #60**
- Addressed **favorited and nearest eateries should by animated in #64**
- Addressed **[UI Enhancement] Eatery collapse animation clunky for eatery not on screen #76**
- Addressed **make open eateries green in upcoming menus screen #62**
- Addressed **tapping on recent search twice loads in two views #92**
- Addressed **swipe left to be able to delete a recent search on search feature #87**
- Addressed **payment methods on home screen doesn’t drop down #81**
- Addressed **add an interaction that if u swipe right, it returns u to the original upcoming menus page #85**
- Addressed **Favorite eatery icon isn't highlighted in search #93**

******************************************************This new AppStore release also addresses the following:******************************************************

- Addressed **more usability issue but for people on unlimited meal plans when they click on the account page, they are shown a majority blank screen. #30**
- Addressed **ability to favorite eatery should also be on the home screen instead of only able to favorite once you click on that specific eatery #32**
- Addressed **"closed until ... " should be a new line instead of text wrapping #59**

### Other Changes

- Upcoming menu now shows hours instead of Eatery status.
- Removed Eatery details button
- Removed “All Campus” filter.
- Changed spacing in upcoming menu.
- Added status to upcoming eateries that are happening today.
- Added skeleton for nearby eateries.
- Removed `gray04` color because this is not a valid color and will crash the app.
- Added scale transform animation when tapping on a cell.
- Sorted the order of the menu to align with Cornell Dining (but with the main entree at the very top).
- Change the font weight of the menu category label.

## Next Steps

- Loading single eateries should be cached. No need to use NSCache here. The `InMemoryCache` should be good enough.
- Empty states for Home Page and Upcoming Menus.
- Better animation throughout the entire app (e.g. tapping on buttons should have some animation curve to it instead of just popping up).
- Backend should remove “Dining Room” for the dining halls as well as “Jansen’s Dining Room at” for Bethe.

## Screenshots

Check on TestFlight.

### CODE REVIEW

File: Eatery Blue/UI/EateryCard/EateryExpandableCardContentView.swift [hunk 1]
```diff
@@ -87,44 +104,22 @@ class EateryExpandableCardContentView: UIView {
     }
     
     private func setupEateryStatusLabel() {
-        eateryStatusLabel.textColor = UIColor.Eatery.black
+        eateryStatusLabel.textColor = UIColor.Eatery.gray03
```
Comment (antoinztte) [inline Eatery Blue/UI/EateryCard/EateryExpandableCardContentView.swift:107]: Did we confirm this with design

File: Eatery Blue/UI/EateryCard/EateryExpandableCardDetailView.swift [hunk 2]
```diff
@@ -75,7 +64,7 @@ class EateryExpandableCardDetailView: UIView {
         menuCategoryStackView.axis = .vertical
         menuCategoryStackView.alignment = .fill
         menuCategoryStackView.distribution = .equalSpacing
-        menuCategoryStackView.spacing = 8
+        menuCategoryStackView.spacing = 12
```
Comment (antoinztte) [inline Eatery Blue/UI/EateryCard/EateryExpandableCardDetailView.swift:67]: What did kathleen say abt this

**Review by antoinztte — APPROVED** [review]

File: Eatery Blue/UI/EateryCard/EateryExpandableCardContentView.swift (hunk 1 above)
Comment (vinnie4k) [reply Eatery Blue/UI/EateryCard/EateryExpandableCardContentView.swift:107]: I asked Liam.

File: Eatery Blue/UI/EateryCard/EateryExpandableCardDetailView.swift (hunk 2 above)
Comment (vinnie4k) [reply Eatery Blue/UI/EateryCard/EateryExpandableCardDetailView.swift:67]: I asked Liam. We can change this if Kathleen wants.

File: Eatery Blue/Model/EateryFormatter.swift [hunk 3]
```diff
@@ -83,15 +109,15 @@ class EateryFormatter {
     func eateryCardFormatter(_ eatery: Eatery, date: Date) -> NSAttributedString? {
         let day = Day(date: date)
         if eatery.isOpen {
-            if let nextEventOfDay = EateryStatus.nextEvent(eatery.events, date: date, on: day) {
+            if case .open(let event) = eatery.status {
```
[the remark this answers is not in the document]
Comment (vinnie4k) [reply Eatery Blue/Model/EateryFormatter.swift:112]: Yea I agree this should be a switch. IDK why Will Ma used this syntax but I'll change it.

**Pull request merged**